app/views/admin/order_cycles/incoming.html.haml

```diff
@@ -1,3 +1,23 @@
+- content_for :page_actions do
+  :javascript
```

Comment:
These duplicated chunks of code are also present in `app/views/admin/order_cycles/edit.html.haml`, right?

In this case (the same view code duplicated 3 times) it would be good to extract that code to a partial and re-use it instead of having the same thing defined in multiple places :+1:

Reply:
Will definitely do that. Thanks for the tip :+1:

Reply:
We can also delete that same chunk of code in `app/views/admin/order_cycles/edit.html.haml` and replace it with the same partial, now that it'[s been extracted.

Reply:
The problem I face is that these three lines of code only work on the `app/views/admin/order_cycles/edit.html.haml` page as it's the only page with routes set up to notify producers. So I have to have this separately, but that renders the notify producers button outside of the box.  
```Ruby
  - if can? :notify_producers, @order_cycle
    %li
      = button_to t(:notify_producers), main_app.notify_producers_admin_order_cycle_path, :id => 'admin_notify_producers', :confirm => t(:are_you_sure)
```

Reply:
You could also have it be like this in the `edit` page 
```Ruby
= render partial: "/admin/order_cycles/order_cycle_top_buttons"
- content_for :page_actions do
  - if can? :notify_producers, @order_cycle
    %li
      = button_to t(:notify_producers), main_app.notify_producers_admin_order_cycle_path, :id => 'admin_notify_producers', :confirm => t(:are_you_sure)
```
If this is acceptable, let me know and I'll push it :smile:

Reply:
Yeah I'd go with that if we can re-use the partial in all three places (assuming it works) :+1:

Reply:
Oh it works, I just wanted to run it by you before I made the commit. Just to make sure.

Reply:
Great, that looks much better!